lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1791,6 +1793,56 @@ class DecomposeAtenToDtypeLayoutOp
 };
 } // namespace
 
+namespace {
+// Decompose the `aten.select_scatter` operation into `aten.slice_scatter` op.
+class DecomposeAtenSelectScatterOp
+    : public OpRewritePattern<AtenSelectScatterOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenSelectScatterOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value start = op.index();
+    Value dim = op.dim();
+    Value self = op.self();
+    Value src = op.src();
+
+    Value one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value startPlusOne =
+        rewriter.create<AtenAddIntOp>(loc, one.getType(), start, one);
+    BaseTensorType srcTensorType = src.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    int64_t dimInt = 0;
+    if (srcTensorType.hasSizes()) {
```

Comment:
Make sure to return early to avoid indentation. In this case I would do

```cpp
if (!srcTensorType.hasSizes())
  return rewriter.notifyMatchFailure(op, "dim is not a valid dim");

// rest of code
```

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code